mesonbuild/dependencies/dev.py

```diff
@@ -255,19 +254,20 @@ def __fix_bogus_link_args(self, args):
               "-L IBPATH:...", if we're using an msvc like compilers convert
               that to "/LIBPATH", otherwise to "-L ..."
         """
-        cpp = self.env.coredata.compilers[self.for_machine]['cpp']
 
         new_args = []
         for arg in args:
             if arg.startswith('-l') and arg.endswith('.so'):
                 new_args.append(arg.lstrip('-l'))
             elif arg.startswith('-LIBPATH:'):
+                cpp = self.env.coredata.compilers[self.for_machine]['cpp']
+                assert isinstance(cpp, CLikeCompiler)
```

Comment:
why is this assert needed?

Reply:
Because `get_linker_search_args` is only defined in `CLikeCompiler` and not in `Compiler` (which is what is stored in `coredata.compilers`).

Reply:
We should fix that, can you add a stub definition to `Compiler`? I've been trying to fix this whenever I find cases like this. Even if the stub implementation is `raise MesonException(f{self.id} doesn't implement this'}` that would be great.

Reply:
No, it's sort of a historical problem. Originally each compiler sublcass just added methods randomly, as we've added type annotations and fixed a ton of those issues I've added them to the base class. Thanks for fixing that!